### ISSUE TITLE
Opening private thread that doesn't include you eats ram and crashes node

### ISSUE THREAD

**Issue by sbillig** [fraction/oasis#334]
## What's the problem you want solved?
Oasis's node process pegs cpu, eats ram, and aborts after clicking on Daan's (encrypted) "private wiki" link.

See this thread and click the top link: http://localhost:3000/thread/%25jwaEMnDl74FsykuI3t8ruRAO%2FpjokcFx3I40P0mE4rk%3D.sha256#%25jwaEMnDl74FsykuI3t8ruRAO%2FpjokcFx3I40P0mE4rk%3D.sha256

Here's one of my private threads; perhaps someone can open this and see if the same thing happens:
http://localhost:3000/thread/%25vrt6fReIeHo5ODdISW17AYMKlnJVVc77ObpKomdtwGw%3D.sha256#%25vrt6fReIeHo5ODdISW17AYMKlnJVVc77ObpKomdtwGw%3D.sha256

```
==== JS stack trace =========================================

    0: ExitFrame [pc: 0x23c81d95be3d]
Security context: 0x37d4b959e6e9 <JSObject>
    1: split [0x37d4b95906c9](this=0x37d434f57819 <String[38]: 512M1.54369203770100000000000000000000>,0x37d49eb888a9 <String[1]: M>)
    2: D [0x37d47f041821] [/usr/local/lib/node_modules/@fraction/oasis/node_modules/charwise/codec/number.js:~26] [pc=0x23c81e07b8d1](this=0x37d41ca3f441 <Object map = 0x37d43dee8e81>,encoded=0x37d434f57549 <String[40]: FE512M...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x10003b540 node::Abort() [/usr/local/bin/node]
 2: 0x10003b719 node::OnFatalError(char const*, char const*) [/usr/local/bin/node]
 3: 0x10017b547 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 4: 0x10017b4e8 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 5: 0x10043133c v8::internal::Heap::UpdateSurvivalStatistics(int) [/usr/local/bin/node]
 6: 0x100432f7b v8::internal::Heap::CheckIneffectiveMarkCompact(unsigned long, double) [/usr/local/bin/node]
 7: 0x100430616 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/usr/local/bin/node]
 8: 0x10042f435 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
 9: 0x100437c16 v8::internal::Heap::AllocateRawWithLigthRetry(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/local/bin/node]
10: 0x100437c60 v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/local/bin/node]
11: 0x10041b3bd v8::internal::Factory::NewProperSubString(v8::internal::Handle<v8::internal::String>, int, int) [/usr/local/bin/node]
12: 0x100620331 v8::internal::Runtime_StringSplit(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/local/bin/node]
13: 0x23c81d95be3d 
fish: 'oasis' terminated by signal SIGABRT (Abort)
```

## Is there a solution you'd like to recommend?

Don't do that.

## What version or commit of Oasis are you using?

2.14.0, node 10.16.0. I'll try to upgrade some things and see if it magically fixes itself.

**Comment by christianbundy** [comment]
That's fantastic, thanks for pointing this out. My guess is that this is going to be some JavaScript baloney where `.value.content.root` is `undefined` and we're doing a bunch of silly queries looking for that message. :grimacing:

**Comment by christianbundy** [comment]
Output with `--debug`:

```
  oasis getting root ancestor of %vrt6fReIeHo5ODdISW17AYMKlnJVVc77ObpKomdtwGw=.sha256 +0ms
  oasis private message +0ms
  oasis not a post +1ms
  oasis got mysterious root ancestor that fails all known schemas +0ms
  oasis {
  oasis   key: '%vrt6fReIeHo5ODdISW17AYMKlnJVVc77ObpKomdtwGw=.sha256',
  oasis   value: {
  oasis     previous: '%p8wWRi5a4Qv4boE6PPPP946QsfeUn+uFsBIlZmnjB5Y=.sha256',
  oasis     sequence: 84,
  oasis     author: '@ZFWw+UclcUgYi081/C8lhgH+KQ9s7YJRoOYGnzxW/JQ=.ed25519',
  oasis     timestamp: 1575494222075,
  oasis     hash: 'sha256',
  oasis     content: 'zg3JRiVyCrLKwWZwjN9SVujl3qK7YTwMUWCxcmK9XudYxiARWg6rxnd/eVpSrE+PT4V/2EO+lwHk0S+/MqTNCVAX2PyXmMqoUmtU6Wq8NSLt8xI6qVzO54v9RP7LtK8h6yM7QVSC7WH01CzBTla9YKjW+dbAxhgsXzy1mkbn0aqb/x9Y58fJ+wjUbbuiYEZRKpEkW09G3F8C4YvmpuQSFvXtbkiLIY0cJldXIVBBJIAOFJkA+oV7+UhF3D8Cf5MRX453pxHamn+nWn+shyLndSHu/ewEFHAnmh+PePSjlkEnSC8JJ7d4vYrDQ6L6u+UWj02DA5qWZ5MKQa3pCe8BnABUwrxYNnV+zKrXz/ga7EcQezqPu6G6cu++2XG8krdtZ1JvH2l5FOwXvcknD3W19lQoQrdCDmrMKR6ErszLY6OX/5rP5d7SkDPROn3/FpGwHvxLbxrc5g2PWUgefrXBhJKFysMft2ilwMrt3PeSVFSvQ26jLMYE9cp6IaDB+9EdAquTS/CE0NsSz9ple1OvTby47Q+J170aC0W4bRuqwfVzEmsK8Zw4+UXnzgdKNHpX9Y43ZflpwI75egU/xbsfeOwr7Flp1hAc+8BTUSZ/60hXkAC23oiGoPsw8zvuw3I3wzYoCwnlxJbcXTJu33RQrYCP2C9Tq14tLuMahBd5Xtp8R8exXiNEFJ8LY/+UbPkilAh6hyD6jZuHyCSet23139E8WAoFA4wGDXAXRu/tnbJNT5LvDtQHHEUK+T1pRNQ0lApXOs0=.box',
  oasis     signature: '2FDeYWev2JXROTE2SBzkHJIC65i+tEp33vV8ZuSHDm0KKGBRSENb1LvL/Kg2OLGrJm2PPUe4tYouzY34F5+JAw==.sig.ed25519'
  oasis   },
  oasis   timestamp: 1575494224231
  oasis } +0ms
  oasis got root ancestors +3ms
```

**Comment by christianbundy** [comment]
Just pushed a fix in #335, if you want to test, but it looks to me like it resolves the problem. :tada:
